cmd/configschema/yamlgen/yamlgen/README.md

```diff
@@ -0,0 +1,102 @@
+# Config Schema YAML Generator (alpha)
+
+This CLI application creates configschema YAML files for collector components
+where each YAML file describes the configuration fields of its corresponding
+component.
+
+## Operation
+
+There are two major modes of operation: one that create YAML files in each
+collector component directory and another that creates a zip archive comprising
+all the collector YAML files.
+
+
+### Create YAML files
+
+This mode of operation creates a `cfgschema.yaml` file per component,
+writing each one to its corresponding component directory in the locally
+checked-out source tree.
+
+
+```yaml
+cd cmd/configschema
+go run ./yamlgen
+```
+
+### Create a Zip archive
+
+This mode of operation, specified by the `-z` flag, creates a zip
+archive comprising all the component YAML files.
+
+```yaml
+cd cmd/configschema/yamlgen
+go run . -d ../../.. -z -f myfile.zip
+```
+
+### Command line arguments
+
+* `-z` create a zip archive
+* `-f <filename>` write the zip archive to filename (for use with `-z`, defaults to `yamlgen.zip`)
+* `-d <directory>` the path to the collector source root directory (defaults to `../..`)
+
+## Example Output
+
+The following is an example `cfgschema.yaml` produced for the File Exporter. 
+
+```yaml
+type: '*fileexporter.Config'
+doc: |
+  Config defines configuration for file exporter.
+fields:
+- name: path
+  kind: string
+  default: ""
+  doc: |
+    Path of the file to write to. Path is relative to current directory.
+- name: rotation
```

Comment:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/15231 doesn't mention how the generated yaml files are expected to be consumed. I though they are meant to be put in components README.md so people can read them and easily copy/paste into their collector configurations if needed. Is the purpose of these yaml files different?

Reply:
Yes, the purpose is different from that. They are metadata files about component configs (field names, types, defaults, docs, etc.), and are intended to be used by tools and/or for generating documentation (that's what our org wants them for). So, I'm not sure they'd be appropriate for reading by humans as-is or for copy/pasting, since they're not actually configs. Since they are metadata, I was thinking that should live next to the `metadata.yaml` files in each component dir. Also, having thought about it a bit more, I'm thinking it'd be better if the files were called `config-metadata.yaml`.

Reply:
I'm going to see if I can bring this up at the next SIG to see what folks think.

Reply:
Sounds good. I believe these files can be potentially used be the contrib repo itself as a intermediary format to generate human readable yamls that can be automatically inserted in README.md

Reply:
That's an interesting idea -- is your suggestion to generate an example yaml config for each component based on the metadata?

Reply:
Yes, but it's not like an example, it's rather a full configuration with default values and docs rendered as comments for each config option